drivers/sensor/lps22df/lps22df_trigger.c

```diff
@@ -0,0 +1,257 @@
+/* ST Microelectronics LPS22DF pressure and temperature sensor
+ *
+ * Copyright (c) 2023 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/lps22df.pdf
+ */
+
+#define DT_DRV_COMPAT st_lps22df
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/logging/log.h>
+
+#include "lps22df.h"
+
+LOG_MODULE_DECLARE(LPS22DF, CONFIG_SENSOR_LOG_LEVEL);
+
+/**
+ * lps22df_enable_int - enable selected int pin to generate interrupt
+ */
+static int lps22df_enable_int(const struct device *dev, int enable)
+{
+	const struct lps22df_config * const cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	lps22df_pin_int_route_t int_route;
+
+	/* set interrupt */
+	lps22df_pin_int_route_get(ctx, &int_route);
+	int_route.drdy_pres = enable;
+	return lps22df_pin_int_route_set(ctx, &int_route);
+}
+
+/**
+ * lps22df_trigger_set - link external trigger to event data ready
+ */
+int lps22df_trigger_set(const struct device *dev,
+			  const struct sensor_trigger *trig,
+			  sensor_trigger_handler_t handler)
+{
+	struct lps22df_data *lps22df = dev->data;
+	const struct lps22df_config * const cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	lps22df_data_t raw_data;
+
+	if (trig->chan == SENSOR_CHAN_ALL) {
```

Comment:
swap the logic here,

```c
if (trig->chan != SENSOR_CHAN_ALL) {
        return -ENOTSUP;
}

<normal execution inline>
```

easies rto track the intended code path

Reply:
seems proper